example/viam_example_app/lib/screens/stream.dart

```diff
@@ -18,14 +22,59 @@ class StreamScreen extends StatefulWidget {
 }
 
 class _StreamScreenState extends State<StreamScreen> {
+  // Stream
   bool _ready = false;
   late RTCVideoRenderer _renderer;
   late async.StreamSubscription<MediaStream> _streamSub;
 
+  // Single frame
+  ByteData? imageBytes;
+  bool _imgLoaded = false;
+
+  void _getImage() {
+    setState(() {
+      _imgLoaded = false;
+    });
+    final imageFut = widget.camera.getImage();
+    imageFut.then((value) {
+      final convertFut = convertImageToFlutterUi(value.image ?? img.Image.empty());
+      convertFut.then((value) {
+        final pngFut = value.toByteData(format: ui.ImageByteFormat.png);
+        pngFut.then((value) => setState(() {
+              imageBytes = value;
+              _imgLoaded = true;
+            }));
+      });
+    });
+  }
+
+  Future<ui.Image> convertImageToFlutterUi(img.Image image) async {
+    if (image.format != img.Format.uint8 || image.numChannels != 4) {
+      final cmd = img.Command()
+        ..image(image)
+        ..convert(format: img.Format.uint8, numChannels: 4);
+      final rgba8 = await cmd.getImageThread();
+      if (rgba8 != null) {
+        image = rgba8;
+      }
+    }
+
+    ui.ImmutableBuffer buffer = await ui.ImmutableBuffer.fromUint8List(image.toUint8List());
+
+    ui.ImageDescriptor id = ui.ImageDescriptor.raw(buffer, height: image.height, width: image.width, pixelFormat: ui.PixelFormat.rgba8888);
+
+    ui.Codec codec = await id.instantiateCodec(targetHeight: image.height, targetWidth: image.width);
+
+    ui.FrameInfo fi = await codec.getNextFrame();
+    ui.Image uiImage = fi.image;
+
+    return uiImage;
+  }
+
   @override
   initState() {
     super.initState();
-    _startStream();
+    // _startStream();
```

Comment:
intended?

Reply:
Yup, I want to start stream when the user presses the button now, not on load. I will remove this line.